variables.tf

```
@@ -361,6 +361,33 @@ variable "advanced_threat_protection_enabled" {
   nullable    = false
 }
 
+variable "azure_files_authentication_directory_type" {
+  description = "Determines if azure_files_authentication should be enabled. Possible values are AD, AADS or AADKERB"
+  type        = string
+  default     = null
+  nullable    = true
+
+  validation {
+    error_message = "The active_directory object is only supported when directory_type is AD"
```

Set relevant error message:

```suggestion
    error_message = "Value of azure_files_authentication_directory_type must be \"AD\", \"AADDS\" , \"AADKERB\" or null."
```